### PR TITLE
 VACMS-6980: Add PDF audit view.

### LINKED ISSUE #6980: PDF accessibility audit tool for VA 508 office
## User Story or Problem Statement

As a 508 office accessibility reviewer...

1. I want a repository/directory/tool with which i can run accessibility reviews for PDFs that have been uploaded to VA.gov
2. I want to know who has added them, and what Section owns them (eg a VAMC system, or other)
3. I want to know what pages these PDFs have been added to, so that i can understand the context 

## Acceptance Criteria
- [x] VA 508 office can access a list of all PDFs that have been added, via CMS, to VA.gov. 
- [x] Ideally this list exists as a view in CMS, but can also be exported by CSV
- [x] Stretch: shows which user uploaded them, and to what Section.
- [ ] Stretch: List shows where these PDFs appear.
- [x] Whatever stretch items do not get finished are added to a [followup issue](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/8014).

Added after the first pass

Seeing this in action helps me think through a few ACs
- [ ] I think the path for view should be /admin/content/media/audit/pdfs, and the one for urls should be /admin/content/media/audit/image-alt, ideally with /admin/content/media as a parent menu item.
- [ ] Direct access to the PDF URLs in audit
  - [ ] https://www.va.gov/files/YYYY-MM//[filename].pdf is listed as column beside the name. I believe YYYY-MM is date created but that probably needs to be double checked. 
  - [ ] https://prod.cms.va.gov/[path-to-pdf] is listed as column
- [ ] Exposed filter label should be "Name contains" not "Name"
- [ ] Section should be linked. (minor but quick)
- [ ] We've validated "number of uses" is helpful, or misleading. [See screenshot in comment on PR](https://github.com/department-of-veterans-affairs/va.gov-cms/pull/8013#pullrequestreview-882140044)

## Implementation steps
<optional>

## Design principles

Veteran-centered
- [ ] `Single source of truth`: Increase reliability and consistency of content on VA.gov by providing a single source of truth.
- [ ] `Accessible, plain language`: Provide guardrails and guidelines to ensure content quality.
- [ ] `Purposely structured content`: Ensure Content API can deliver content whose meaning matches its structure.
- [ ] `Content lifecycle governance`: Produce tools, processes and policies to maintain content quality throughout its lifecycle.

Editor-centered
- [ ] `Purpose-driven`: Create an opportunity to involve the editor community in VA’s mission and content strategy goals.
- [ ] `Efficient`: Remove distractions and create clear, straightforward paths to get the job done.
- [ ] `Approachable`: Offer friendly guidance over authoritative instruction.
- [ ] `Consistent`: Reduce user’s mental load by allowing them to fall back on pattern recognition to complete tasks.
- [ ] `Empowering`: Provide clear information to help editors make decisions about their work.

### CMS Team

Please leave only the team that will do this work selected. If you're not sure, it's fine to leave both selected.

- [ ] `Platform CMS Team`
- [x] `Sitewide CMS Team`

### PR DESCRIPTION
**Pull request by msbtterswrth**
## Description

Closes #6980 .

## Testing done
Visited view page and searched for PDFs.

## Screenshots
image

## QA steps

As a CMS admin, 
1. navigate to the new PDF audit (`/admin/content/pdf-audit`)
2. View the list of PDFs within the CMS
3. Then validate Acceptance Criteria from issue
- [x] VA 508 office can access a list of all PDFs that have been added, via CMS, to VA.gov.
- [x] Ideally this list exists as a view in CMS, but can also be exported by CSV
- [x] Stretch: shows which user uploaded them, and to what Section.

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Platform CMS Team`
- [ ] `Sitewide CMS Team`
  - [x] `⭐️ Content ops`
  - [ ] `⭐️ CMS Experience`
  - [ ] `⭐️ Offices`
  - [ ] `⭐️ Product Support`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Review by kevwalsh — CHANGES_REQUESTED** [review]
Nice!

Seeing this in action helps me think through a few ACs
- [x] I think the path for view should be /admin/content/media/audit/pdfs, and the one for urls should be /admin/content/media/audit/image-alt, ideally with /admin/content/media as a parent menu item.
- [x] Direct access to the PDF URLs in audit
  - [ ] https://www.va.gov/files/YYYY-MM//[filename].pdf is listed as column beside the name. I believe YYYY-MM is date created but that probably needs to be double checked. 
  - [x] https://prod.cms.va.gov/[path-to-pdf] is listed as column
- [x] Exposed filter label should be "Name contains" not "Name"
- [x] Section should be linked. (minor but quick)

I'm also wondering if we need to investigate some of this data to see if "number of uses" is helpful, or misleading. See screenshot below. I'm guessing only one of each of these is actually available on www.va.gov.

<details><summary>Screenshot</summary>

PDF_Audit___VA_gov_CMS


</details>

File: composer.json [hunk 1]
```diff
@@ -57,6 +57,7 @@
         "drupal/entity_diff_ui": "^1.0",
         "drupal/entity_embed": "^1.0",
         "drupal/entity_field_fetch": "^1.0@beta",
+        "drupal/entity_media_usage": "^1.0",
```
Comment (kevwalsh) [inline composer.json:60]: I'd defer to Steve on this one, but i don't think we want to add a module if we're not enabling it.

File: config/sync/views.view.pdf_audit.yml [hunk 2]
```diff
@@ -0,0 +1,1566 @@
...
+  module:
+    - csv_serialization
+    - file
+    - media
+    - rest
+    - serialization
+    - user
+    - views_data_export
+    - workbench_access
+id: pdf_audit
```
Comment (kevwalsh) [inline config/sync/views.view.pdf_audit.yml:17]: Should we be putting all the media audits into one view? then the machine name should be media_audit, and we would put the alt text audit here too. Not a blocker for me, just a q.

File: config/sync/views.view.pdf_audit.yml [hunk 3]
```diff
@@ -0,0 +1,1566 @@
...
+            remember: false
+            default_group: All
+            default_group_multiple: {  }
+            group_items: {  }
+          fields:
+            uri: uri
+          plugin_id: combine
+      sorts: {  }
+      title: 'PDF Audit'
+      header: {  }
```
Comment (kevwalsh) [inline config/sync/views.view.pdf_audit.yml:861]: Can we put the Number of results component, which is standard on most of our views?

File: config/sync/views.view.pdf_audit.yml [hunk 4]
```diff
@@ -0,0 +1,1566 @@
...
+          table: views
+          field: nothing
+          relationship: none
+          group_type: group
+          admin_label: ''
+          label: 'File Link'
+          exclude: false
+          alter:
+            alter_text: true
+            text: "{% set public= 'public://' %}\r\n{% set local= '/' %} \r\n<a href=\"{{ uri|replace({ (public): local}) }}\">{{ uri|replace({ (public): local}) }}</a>"
```
Comment (kevwalsh) [inline config/sync/views.view.pdf_audit.yml:1086]: I like this approach, if we are going to include web PDFs.
But 
- [ ] "Local" should be "CMS". 
- [ ] "Prod" should be "Web" or "Prod Web" or "www.va.gov".  (not sure which)

**Review by kevwalsh — CHANGES_REQUESTED** [review]
A couple things, here's a screenshot of the UI bits. 
Cursor_and_PDF_Audit___VA_gov_CMS

File: config/sync/views.view.pdf_audit.yml (hunk 3 above)
Comment (msbtterswrth) [reply config/sync/views.view.pdf_audit.yml:861]: added!

File: config/sync/views.view.pdf_audit.yml (hunk 4 above)
Comment (msbtterswrth) [reply config/sync/views.view.pdf_audit.yml:1086]: I went with 'Web', that seemed to be the most plain language version of those options.

File: composer.json (hunk 1 above)
Comment (msbtterswrth) [reply composer.json:60]: good catch, I thought I got them both!

File: config/sync/views.view.pdf_audit.yml (hunk 2 above)
Comment (msbtterswrth) [reply config/sync/views.view.pdf_audit.yml:17]: We have another ticket for for improving views availability (#6759) in which I suggest a lot of path changes and reorganization. I think we could think through where everything lives holistically there.

**Review by swirtSJW — COMMENTED** [review]
QA step didn't match.  The view is here.
/admin/content/media/audit/pdfs

"Last changed on" ,    "Updated" is the pattern on other views.

**Review by swirtSJW — APPROVED** [review]

**Pull request merged**